src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/CLHSDB.java

```diff
@@ -106,8 +107,8 @@ public void reattach() {
             };
 
 
-        BufferedReader in =
-            new BufferedReader(new InputStreamReader(System.in));
+        Charset charset = Charset.forName(System.getProperty("stdin.encoding"), Charset.defaultCharset());
+        BufferedReader in = new BufferedReader(new InputStreamReader(System.in, charset));
```

Comment:
Why in some cases are you special casing the the default charset as you do here, but in other cases you are not? Also, the CharSet.forName() API you are using here is new in 18, which means backports will need to address that.